app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/NavigationOverlayFragment.kt

```diff
@@ -257,6 +261,39 @@ class NavigationOverlayFragment : Fragment() {
         activity!!.moveTaskToBack(true)
     }
 
+    private fun observeAccountState(): Disposable {
+        val fxaRepo = serviceLocator.fxaRepo
+
+        return fxaRepo.accountState.subscribe {
+            when (it) {
+                FxaRepo.AccountState.AUTHENTICATED_WITH_PROFILE -> {
+                    fxaRepo.accountManager.accountProfile()?.let { profile ->
+                        if (profile.avatar != null) {
+                            profile.avatar?.url?.let { url ->
+                                PicassoWrapper.client
+                                    .load(url)
+                                    .into(fxaButton)
+                            }
+                        } else {
+                            fxaButton.setImageResource(R.drawable.ic_avatar_authenticated_no_picture)
+                            fxaButton.contentDescription = resources.getString(R.string.fxa_navigation_item_signed_in2)
+                        }
+                    }
+                }
+                FxaRepo.AccountState.AUTHENTICATED_NO_PROFILE -> {
+                    fxaButton.setImageResource(R.drawable.ic_avatar_authenticated_no_picture)
+                    fxaButton.contentDescription = resources.getString(R.string.fxa_navigation_item_signed_in2)
+                }
+                FxaRepo.AccountState.NEEDS_REAUTHENTICATION,
+                FxaRepo.AccountState.NOT_AUTHENTICATED -> {
+                    fxaButton.setImageResource(R.drawable.ic_fxa_login)
+                    fxaButton.contentDescription = resources.getString(R.string.fxa_navigation_item_new)
```

Comment:
We need to update the hover-over hint too, right?

Reply:
@mcomella Did you mean the tooltip? They appear to be set with `contentDescription` - https://github.com/mozilla-mobile/firefox-tv/blob/master/app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/ToolbarUiController.kt#L68

Reply:
I was not aware, sgtm.

Reply:
Not necessary to change but I wonder if we should make that more literal, e.g. a function like, `setTooltipAndContentDescription`, instead of just setting the content description.